### PR TITLE
[3.11] Update NPM dependencies

### PR DESCRIPTION
**Pull request by pluma4345**
### Scope & Purpose

This updates various NPM dependencies to the most recent compatible version.

This resolves [CVE-2022-39353](https://github.com/advisories/GHSA-crh6-fp67-6883) in `@xmldom/xmldom` as well as three CVEs in dependencies of `eslint`. These are all development dependencies which should not affect the release builds.

- [x] :hankey: Bugfix

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [x] :books: documentation written (release notes, API changes, ...)
- [x] Backports
  - [x] Backport for 3.10: #18201
  - [x] Backport for 3.9: #18202
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,18 @@
 devel
 -----
 
+* Updated internal JavaScript dependencies:
+
+  - @xmldom/xmldom: 8.6.0 -> 0.8.6
```
Comment (jsteemann) [inline CHANGELOG:6]: It looks like we are going from version number 8.6 down to 0.8.6.
Pretty sure there is a release for this, but it looks weird.

**Review by jsteemann — APPROVED** [review]
LGTM in general, except the weird version number in the CHANGELOG

**Comment by dothebart** [comment]
netstat output in cluster runs seems to be working - so LGTM from me.

**Review by dothebart — APPROVED** [review]

**Review by dothebart — APPROVED** [review]

File: js/node/node_modules/.package-lock.json [hunk 2]
```diff
@@ -3,38 +3,39 @@
...
-        "mime-types": "~2.1.24",
-        "negotiator": "0.6.2"
+        "mime-types": "~2.1.34",
+        "negotiator": "0.6.3"
       },
       "engines": {
         "node": ">= 0.6"
       }
     },
     "node_modules/accepts/node_modules/negotiator": {
```
Comment (hkernbach) [inline js/node/node_modules/.package-lock.json:27]: This one is not mentioned in the Changelog entry.

**Review by hkernbach — APPROVED** [review]
LGTM.

**Review by KVS85 — APPROVED** [review]

**Pull request merged**